### PR TITLE
Add mex interface.

### PR DESCRIPTION
**Pull request by twadleigh**
This adds the ability to call julia from matlab through a mex function.

- [x] Implement basic functionality.
- Test basic functionality on:
    - [x] Windows,
    - [ ] Mac OS X,
    - [ ] *nix.
- [x] Automate the build of the mex file.
- [x] Write documentation.
- [ ] Add some automated tests.

This supercedes #36.

### CODE REVIEW

**Comment by tkelman** [comment]
If you were to download matlab compiler runtime onto the travis vm's, that wouldn't come with the ability to create mex files would it?

**Comment by twadleigh** [comment]
@tkelman, I just checked: the matlab runtime comes with all the DLLs and headers that you would, at a minimum, need to build a MEX file. It also includes a `mex` exectuable for compiler/platform-independent(ish) building.

File: src/mex.jl [hunk 1]
```diff
@@ -0,0 +1,52 @@
+export mex_init, mex_args, mex_return, mex_showerror, mex_eval, mex_call
```
Comment (twadleigh) [inline src/mex.jl]: I need to wrap this in a `MEX` module and get rid of the superfluous `mex_` prefix everywhere

**Comment by twadleigh** [comment]
In fact, I think I've decided to put `jlcall` in its own repo, as it can be made to be independent of `MATLAB.jl`.

**Comment by timholy** [comment]
Might be quite interesting to submit this to the FileExchange, and see what MathWorks does with it. If you sell it as "here's a way of enhancing Matlab, by expanding the languages that Matlab can call," it just might fly :smile:.

**Comment by twadleigh** [comment]
I have decided to move all of this functionality into the [`jlcall`](https://github.com/twadleigh/jlcall) project.

**Pull request closed without merging**